src/WebJobs.Script/Utility.cs

```diff
@@ -996,6 +997,59 @@ public static FunctionAppContentEditingState GetFunctionAppContentEditingState(I
             }
         }
 
+        /// <summary>
+        /// Returns scale monitors and target scalers we want to use based on the configuration.
+        /// Scaler monitor will be ignored if a target scaler is defined in the same extensions assembly and TBS is enabled.
+        /// </summary>
+        /// <param name="environment">Environment variables.</param>
+        /// <param name="hostingConfiguration">Housing configuration.This is used to enable TDS on stamp level for specific triggers.</param>
+        /// <param name="scaleMonitors">Registered scale monitors.</param>
+        /// <param name="targetScalers">Registered target scalers.</param>
+        /// <param name="scaleMonitorsToProcess">Scale monitor to process.</param>
+        /// <param name="targetScalersToProcess">Target scaler to process.</param>
+        public static void GetScaleInstancesToProcess(
+            IEnvironment environment,
+            IFunctionsHostingConfiguration hostingConfiguration,
+            IEnumerable<IScaleMonitor> scaleMonitors,
+            IEnumerable<ITargetScaler> targetScalers,
+            out List<IScaleMonitor> scaleMonitorsToProcess,
+            out List<ITargetScaler> targetScalersToProcess)
+        {
+            scaleMonitorsToProcess = new List<IScaleMonitor>();
+            targetScalersToProcess = new List<ITargetScaler>();
+
+            string value = environment.GetEnvironmentVariableOrDefault(EnvironmentSettingNames.TargetBaseScalingEnabled, "0");
+
+            // Check if TBS enabled on app level
+            if (value == "1")
+            {
+                foreach (var scaler in targetScalers)
+                {
+                    string flag = hostingConfiguration.GetValue(scaler.GetType().Name, null);
```

Comment:
I really think the hosting config for targetScalers should be at the assembly level, to match the assembly level check we're doing below. That keeps things in sync with the below, and keeps us out of the business of depending on implementation class names which technically could change since they're an internal implementation detail.

Reply:
You are saying that we should use assembly name as the key in the configuration? @chiangvincent, it looks like we need another change in the SC.

Reply:
Yes